api/app/controllers/concerns/spree/api/v2/storefront/order_concern.rb

```diff
@@ -0,0 +1,54 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        module OrderConcern
+          private
+
+          def render_order(result)
+            if result.success?
+              render_serialized_payload { serialized_current_order }
+            else
+              render_error_payload(result.error)
+            end
+          end
+
+          def ensure_order
+            raise ActiveRecord::RecordNotFound if spree_current_order.nil?
+          end
+
+          def order_token
+            request.headers['X-Spree-Order-Token'] || params[:order_token]
+          end
+
+          def spree_current_order
+            @spree_current_order ||= find_spree_current_order
+          end
+
+          def find_spree_current_order
+            Spree::Orders::FindCurrent.new.execute(
+              store: spree_current_store,
+              user: spree_current_user,
+              token: order_token,
+              currency: current_currency
+            )
+          end
+
+          def serialize_order(order)
+            dependencies[:cart_serializer].new(order.reload, include: resource_includes, fields: sparse_fields).serializable_hash
+          end
+
+          def serialized_current_order
+            serialize_order(spree_current_order)
+          end
+
+          def dependencies
```

Comment:
@przemosk we cannot have dependencies set in the concern - it will overwrite controller dependencies

Reply:
i changed that method name to `serializer`